pkg/services/sqlstore/replstore.go

```diff
@@ -62,40 +77,46 @@ func ProvideServiceWithReadReplica(primary *SQLStore, cfg *setting.Cfg,
 	// by that mimic the functionality of how it was functioning before
 	// xorm's changes above.
 	xorm.DefaultPostgresSchema = ""
-	s, err := newReadOnlySQLStore(cfg, features, bus, tracer)
+
+	// Parsing the configuration to get the number of repls
+	replCfgs, err := NewRODatabaseConfigs(cfg, features)
 	if err != nil {
 		return nil, err
 	}
-	s.features = features
-	s.tracer = tracer
+	replStore.repls = make([]*SQLStore, len(replCfgs))
```

Comment:
In case this makes sense :D 

If `replCfgs` has zero length (can it happen?), `ReadReplica()...rs.nextRepl()` may panic as it only checks rs for nil earlier. 
Or if there would be scenarios with repls initialized to:
```
&ReplStore{
		SQLStore: ss,
		repls:    make([]*SQLStore, 0),        // empty, not nil 
		next:     0,
	}
```
Is it worth updating the [check in ReadReplica](https://github.com/grafana/grafana/pull/90530/files#diff-1d7735e8dccca0373940f5d09d591f096bca4718106498b266e451492affa9d1R42) for [] ?

Reply:
Thanks, that's a good catch! the previous caller would catch if the cfg is nil, but not the repl config